src/core/main.c

```diff
@@ -103,6 +104,7 @@ static bool arg_switched_root = false;
 static bool arg_no_pager = false;
 static bool arg_service_watchdogs = true;
 static char ***arg_join_controllers = NULL;
+static int arg_default_log_max_level = LOG_INFO;
```

Comment:
actually, why do we need this variable at all? i mean, we can just drop it altogether, no, and let config_parse_level2() in place and parse directly into log.c's log level setting. I mean, this early there's really no reason to maintain a shadow field. It's entirely sufficient that as soon manager_new() runs we then store the log level enforced at that point into the shadow variable in the Manager object. 

That would be much simpler and less invasive.

Reply:
It seems that you're missing the reload path. How is this going to work when PID1 is asked to reload its configs ?